src/sentry/utils/circuit_breaker.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+
+from django.core.cache import cache
+
+from sentry.utils import metrics
+
+DEFAULT_ERROR_LIMIT = 30
+ERROR_COUNT_CACHE_KEY = lambda key: f"circuit_breaker:{key}-error-count"
+PASSTHROUGH_COUNT_CACHE_KEY = lambda key: f"circuit_breaker:{key}-passthrough"
+
+
+def circuit_breaker_activated(
+    key: str,
+    error_limit: int = DEFAULT_ERROR_LIMIT,
+    passthrough_limit: int | None = None,
+) -> bool:
+    """
+    Activates the circuit breaker if the error count for a cache key exceeds the error limit.
+
+    The circuit breaker can allow a certain number of requests to pass through per minute, defined by
+    the passthrough limit if provided.
+    """
+    failure_count = cache.get(ERROR_COUNT_CACHE_KEY(key), 0)
+    if failure_count < error_limit:
+        return False  # not blocked
+
+    # Limit has been exceeded, check if we should allow any requests to pass through
+    if passthrough_limit:
+        passthrough_count, start_time = cache.get(
+            PASSTHROUGH_COUNT_CACHE_KEY(key), (0, datetime.now())
+        )
+
+        # If the last event allowed through was more than a minute ago, reset the clock
+        if (datetime.now() - start_time).total_seconds() > 60:
```

Comment:
@snigdhas can we make the 60 seconds configurable? I think that 60 seconds might be too long, maybe 10 seconds is better?

Reply:
can definitely make it configurable - why would 10 seconds be better? 60 seconds gives us more granularity, or do you mean that we would be able to check if Seer is back every 10s instead of 60s?

Reply:
@snigdhas Yes, exactly. 60 seconds is a long time for a circuit breaker.